### PR TITLE
 Add functionality to reorder pages with keyboard

### PR DESCRIPTION
**Pull request by Thomasvdh92**
https://github.com/wagtail/wagtail/issues/5410

Added functionality to rearrange pages in the wagtail admin for keyboard-only users. To do this, you have to _tab_ through the elements after enabling ordering. Then select an element in the drop and drag area by pressing the enter key. You then selected the object and can use the up and down arrow keys to rearrange them. You can press the up or down arrow keys multiple times so you can place a page n amount of times higher or lower. Then press enter again to confirm and the page will be set accordingly in the list. You can deselect an object by pressing the escape key. 

This functionality uses the same backend call as drag and drop does.

### CODE REVIEW

**Comment by Thomasvdh92** [comment]
resolves #5410

File: wagtail/admin/templates/wagtailadmin/pages/listing/_list.html [hunk 1]
```diff
@@ -48,7 +48,7 @@
                 {% page_permissions page as page_perms %}
                 <tr {% if ordering == "ord" %}id="page_{{ page.id|unlocalize }}" data-page-title="{{ page.get_admin_display_title }}"{% endif %} class="{% if not page.live %}unpublished{% endif %} {% block page_row_classname %}{% endblock %}">
                     {% if show_ordering_column %}
-                        <td class="ord">{% if orderable and ordering == "ord" %}<div class="handle icon icon-grip text-replace">{% trans 'Drag' %}</div>{% endif %}</td>
+                        <td class="ord">{% if orderable and ordering == "ord" %}<div tabindex="0" id="orderer" class="handle icon icon-grip text-replace">{% trans 'Drag' %}</div>{% endif %}</td>
```
Comment (zerolab) [inline wagtail/admin/templates/wagtailadmin/pages/listing/_list.html]: Multiple elements with the same id are not valid HTML. Could this not be replaced by a class?

Comment (thibaudcolas) [reply wagtail/admin/templates/wagtailadmin/pages/listing/_list.html]: `data-` attribute if this is only used for JS interaction – 
```suggestion
                        <td class="ord">{% if orderable and ordering == "ord" %}<div tabindex="0" data-orderer class="handle icon icon-grip text-replace">{% trans 'Drag' %}</div>{% endif %}</td>
```

**Comment by lb-** [comment]
@PaarthAgarwal if you're up for it. Might be worth resurrecting this PR - either as a rebase or just apply the changes on main as a new branch. 

Looks like it was close and a good addition. 

I've added to the UX Unification project.

**Comment by PaarthAgarwal** [comment]
> @PaarthAgarwal if you're up for it. Might be worth resurrecting this PR - either as a rebase or just apply the changes on main as a new branch.
> 
> Looks like it was close and a good addition.
> 
> I've added to the UX Unification project.

Sure. You've been working yourself on adding keyboed shortcuts. Does this PR's way of doing it goes along with yours?

**Comment by lb-** [comment]
Good thinking @PaarthAgarwal  - however this keyboard behaviour is more about 'control' of the interaction and not really global keyboard shortcuts.

Have a read, in full, of the issue https://github.com/wagtail/wagtail/issues/5410 to understand the goals here.

File: wagtail/admin/templates/wagtailadmin/pages/listing/_list.html (hunk 1 above)
Comment (lb-) [inline wagtail/admin/templates/wagtailadmin/pages/listing/_list.html:51]: @PaarthAgarwal  what if this was a `button` with `type="button"` then we do not need make it an accessible target with tabindex.

Yeah the order thing will need to be fixed but the data-attribute suggestion above makes sense.

**Comment by lb-** [comment]
@PaarthAgarwal  to get this over the line we would also need to tell screen readers that this can work with the keyboard.

A simple way to do this is a message content that is in the DOM near the top of the table that explains 'focus on blah, press enter to activate, up / down'.

**Comment by PaarthAgarwal** [comment]
> @PaarthAgarwal to get this over the line we would also need to tell screen readers that this can work with the keyboard.
> 
> A simple way to do this is a message content that is in the DOM near the top of the table that explains 'focus on blah, press enter to activate, up / down'.

Should I add a screen reader only message?

**Comment by lb-** [comment]
Yes please Paarth, here is a rough version of the text.

> Press 'enter' to select an item, use up and down arrows to move the item, press 'enter' to deselect the item.

Note: Check that matches the actual behaviour. We can refine further on the PR when up.

**Comment by lb-** [comment]
@Thomasvdh92 thanks for this PR - I know it has been a while but Paarth has taken this on board, finished it off and rebuilt via #8972 so I will close this now.

We really appreciate the original work done here and will add you to the contributors list for this feature.

**Pull request closed without merging**